src/compat/querier.js

```diff
@@ -0,0 +1,144 @@
+'use strict'
+
+const assert = require('assert')
+const EE = require('events')
+const MDNS = require('multicast-dns')
+const Multiaddr = require('multiaddr')
+const PeerInfo = require('peer-info')
+const PeerId = require('peer-id')
+const log = require('debug')('libp2p:mdns:compat:querier')
+const { SERVICE_TAG_LOCAL, MULTICAST_IP, MULTICAST_PORT } = require('./constants')
+
+class Querier extends EE {
+  constructor (peerId, options) {
+    super()
+    assert(peerId, 'missing peerId parameter')
+    this._peerIdStr = peerId.toB58String()
+    this._options = options || {}
+    this._options.queryPeriod = this._options.queryPeriod || 5000
+    this._onResponse = this._onResponse.bind(this)
+  }
+
+  start (callback) {
+    this._handle = periodically(() => {
+      // Create a querier that queries multicast but gets responses unicast
+      const querier = MDNS({ multicast: false, interface: '0.0.0.0', port: 0 })
+
+      querier.on('response', this._onResponse)
```

Comment:
I think having this event listener will prevent the querier objects from being garbage collected
(looking at the [source code](https://github.com/mafintosh/multicast-dns/blob/master/index.js#L114) for mdns, it doesn't seem like `destroy()` removes all event listeners)